### PR TITLE
DC-512: Support UDP protocol in minion

### PR DESCRIPTION
**Pull request by freddy61025**
## Description

- add a runInPlace.sh in minion/assembly for easy startup minion standalone.
- add UDP port expose in install-local/install-local-external-minion-test/docker-compose.yml
- fix flow listener and make sure flow data will reach kafka 

## Jira link(s)
- https://opennms.atlassian.net/browse/DC-512

## Checklist
* [x] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [x] Appropriate reviewer(s) have been selected.
* [x] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [x] Documentation has been updated as necessary.

### CODE REVIEW

File: minion/assembly/runInPlace.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
```
Comment (patrick-schweizer) [inline minion/assembly/runInPlace.sh:1]: NICE!

File: minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/parser/FlowSinkModule.java [hunk 2]
```diff
@@ -45,11 +45,10 @@ public class FlowSinkModule implements SinkModule<TelemetryMessage, TelemetryMes
 
     private final IpcIdentity identity;
 
-    private final String id;
+    private final String id = "Flow";
```
Comment (patrick-schweizer) [inline minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/parser/FlowSinkModule.java:48]: final and not used in constructor: can be static?

**Review by patrick-schweizer — APPROVED** [review]
Looks good to me. Only one nit.

File: minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/parser/ConfigManager.java [hunk 3]
```diff
@@ -63,7 +63,7 @@ public Listener create(Consumer<ServiceMonitorResponse> resultProcessor, Any con
         try {
             this.flowsConfig = config.unpack(FlowsConfig.class);
             holder.clear();
-            flowsConfig.getListenersList().stream().map(telemetryRegistry::getListener);
+            flowsConfig.getListenersList().stream().forEach(telemetryRegistry::getListener);
```
Comment (geraldhumphries) [inline minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/parser/ConfigManager.java:66]: nit: might be able to just do `flowsConfig.getListenersList().forEach(...`

**Review by geraldhumphries — APPROVED** [review]

File: minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/listeners/factory/TcpListenerFactory.java [hunk 4]
```diff
@@ -70,6 +72,15 @@ public FlowsListener createBean(ListenerConfig listenerConfig) {
         if (parser.size() != listenerConfig.getParsersCount()) {
             throw new IllegalArgumentException("Each parser must be of type TcpParser but was not.");
         }
-        return new TcpListener(listenerConfig.getName(), parser.iterator().next(), new MetricRegistry());
+        int port = 0;
+        try {
+            Optional<Parameter> parameter = listenerConfig.getParametersList().stream().filter(p -> "port".equals(p.getKey())).findFirst();
```
Comment (geraldhumphries) [inline minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/listeners/factory/TcpListenerFactory.java:77]: This is not a review comment. I wanted to try some Optional-based code golf:

```        
int port = listenerConfig.getParametersList().stream()
            .filter(p -> "port".equals(p.getKey()))
            .findFirst()
            .map(parameter -> {
                try {
                    return Integer.parseUnsignedInt(parameter.getValue());
                } catch (NumberFormatException e) {
                    throw new IllegalArgumentException(String.format("Invalid port for listener: %s, error: %s", listenerConfig.getName(), e.getMessage()));
                }
            })
            .orElse(0);
```

This isn't a suggestion either, unless you really like the Stream API!

File: minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/parser/ConfigManager.java (hunk 3 above)
Comment (freddy61025) [reply minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/parser/ConfigManager.java]: done

File: minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/listeners/factory/TcpListenerFactory.java (hunk 4 above)
Comment (freddy61025) [reply minion/minion-flows/flows-parser/src/main/java/org/opennms/horizon/minion/flows/listeners/factory/TcpListenerFactory.java:77]: I think we will extract port out from parameter list when we start to enhance flow config in next phase.

**Pull request merged**